zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RequestMapper.java

```diff
@@ -373,14 +375,21 @@ public static Either<ProblemDetail, BroadcastSignalRequest> toBroadcastSignalReq
   public static Authentication getAuthentication() {
     final List<String> authorizedTenants = TenantAttributeHolder.tenantIds();
 
-    final String token =
+    final var token =
         Authorization.jwtEncoder()
             .withIssuer(JwtAuthorizationBuilder.DEFAULT_ISSUER)
             .withAudience(JwtAuthorizationBuilder.DEFAULT_AUDIENCE)
             .withSubject(JwtAuthorizationBuilder.DEFAULT_SUBJECT)
-            .withClaim(Authorization.AUTHORIZED_TENANTS, authorizedTenants)
-            .encode();
-    return new Builder().token(token).tenants(authorizedTenants).build();
+            .withClaim(Authorization.AUTHORIZED_TENANTS, authorizedTenants);
+
+    final var authenticatedUser =
+        (CamundaUser) SecurityContextHolder.getContext().getAuthentication().getPrincipal();
+
+    if (authenticatedUser != null) {
+      token.withClaim(Authorization.AUTHORIZED_USER_KEY, authenticatedUser.getUserKey());
+    }
```

Comment:
❌  I believe we can make this method return an `Either<ProblemDetail, Authentication>`. If authorization is enabled and we don't have an authenticated user we can fail early with a 403 Forbidden. That makes more sense than sending it to Zeebe who will respond with a 401 Unauthorized.

Reply:
I can live with that for now. I do think we should improve it later on 👍